pkg/operator/discover/discover.go

```diff
@@ -190,18 +194,76 @@ func (d *Discover) createDiscoverDaemonSet(namespace, discoverImage string) erro
 func ListDevices(context *clusterd.Context, namespace, nodeName string) (map[string][]sys.LocalDisk, error) {
```

Comment:
Do we really want to wait for the devices? It seems like when the cluster crd is created we should go ahead and create osds and not wait. Then we can watch the configmap in case more devices are discovered. Adding a watch could be part of a future PR, but it doesn't seem right to wait in this PR either.

Reply:
after thinking about this more, we probably do need to wait for the startup scenario. The discovery pod might take some time to initialize even before they have created a crd. Do we at least know for sure that the discovery pod is expected to run on this requested node? Otherwise we might wait for no reason.

Reply:
i have this for extra insurance. So far I can get devices without waiting. I can reduce the max wait.

Reply:
The question is if we assume the best case when OSDs are deployed that the node a) responds in general fast to the commands used here and b) the node is not overloaded.